pkg/koordlet/metricsadvisor/collectors/podresource/pod_resource_collector.go

```diff
@@ -181,6 +196,7 @@ func (p *podResourceCollector) collectPodResUsed() {
 func (p *podResourceCollector) collectContainerResUsed(meta *statesinformer.PodMeta) {
 	klog.V(6).Infof("start collectContainerResUsed")
 	pod := meta.Pod
+	containerMetrics := make([]metriccache.MetricSample, 0)
```

Comment:
The same as the `podMetrics`. It seems that the variable should be newed inside the loop.